Assets/Scripts/Model/Unit.cs

```diff
@@ -72,4 +67,10 @@ void OnTriggerExit (Collider col)
 			this.weapon.removeTarget (col.gameObject.GetComponent<CanReceiveDamage> ());
 		}
 	}
+
+	public bool isDead ()
+	{
+		return this == null;
```

Comment:
> This typically occurs when you call Object.Destroy on something to destroy it in unmanaged memory, and then try to access the object's properties afterwards.

Literally what you're doing here, aren't you? xD Because you Destroyed the object when the projectile killed it and you're calling one if its properties (isDead) after Destroy.

Isn't it easier and safer to check for null outside the object?